hedgehog/src/Hedgehog/Internal/Gen.hs

```diff
@@ -1193,8 +1193,33 @@ frequency = \case
       total =
         sum (fmap fst xs0)
 
-    n <- integral $ Range.constant 1 total
-    pick n xs0
+      upTo n = \case
+        [] ->
+          []
+        (k, x) : xs ->
+          if n - k <= 0 then
+            []
+          else
+            x : upTo (n - k) xs
+
+    -- Pick a single generator from the weighted list of generators. We prune
+    -- here as shrinks no longer need to care about bias.
+    n <- prune $ integral $ Range.constant 1 total
```

Comment:
Instead of `integral` and `prune`, just use `integral_`.

Reply:
Thanks, I didn't know about that!